### ISSUE TITLE
deprecate rule `prettier` since `eslint-plugin-prettier` itself supports virtual filenames correctly now

### ISSUE THREAD

**Issue by JounQin** [dotansimha/graphql-eslint#395]
See

https://github.com/prettier/eslint-plugin-prettier/pull/401/files#diff-e9a90fe7b2d3420b8e5f64e24d5f56ffe6383e47b07461172cc29ed21b768905R197-R199

and

https://github.com/prettier/eslint-plugin-prettier/blob/master/eslint-plugin-prettier.js#L192-L209

---

Besides, actually it is supported previously too. All the user need to do is like the following:

```js
module.exports = {
  // ...
  overrides: [
    {
      files: '*.graphql', // it includes virtual filenames like `*.js/0_document.graphql` already
      rules: {
        'prettier/prettier': [
          2,
          {
            parser: 'graphql',
          },
        ],
      },
    },
  ],
}
```

<del>
After `eslint-plugin-prettier@3.4.0`, the above configuration is not needed any more too!
</del>

Maybe in `eslint-plugin-pretter@3.5.0` after https://github.com/prettier/eslint-plugin-prettier/pull/413 and https://github.com/prettier/eslint-plugin-prettier/pull/415 been merged and released.

**Comment by Urigo** [comment]
@JounQin that's awesome, thank you for pointing that out!

Would you be willing to send a PR for that?
Maybe also worth updating the docs?  because it might be breaking for someone who uses the existing rule?
(and thank you again for your previous PR!)

**Comment by JounQin** [comment]
I'll raise a PR for it.

But I just found the `eslint-plugin-prettier` decided to be compatible with `eslint-plugin-graphql` which is not compatible with this plugin before. So the option for `prettier` and non-virtual filenames is still required, I think we can add some configs which includes this option automatically.

And it seems `eslint-plugin-grahql` is out of maintainace, so I'm going to ask `eslint-plugin-prettiet` to remove the related codes.

**Comment by JounQin** [comment]
I have an idea to support the both two ESLint grahql plugins for `eslint-plugin-prettier` just now, I'll PR there first.

**Comment by Urigo** [comment]
@JounQin yes I agree they should remove `eslint-plugin-grahql`, it's unmaintained for many years now

**Comment by Urigo** [comment]
reference for why we chose not to go with `eslint-plugin-grahql`: https://the-guild.dev/blog/introducing-graphql-eslint

**Comment by JounQin** [comment]
I've raised https://github.com/prettier/eslint-plugin-prettier/pull/413 to make sure `eslint-plugin-prettier` will be compatible with this plugin out of the box.

I'll raise another PR to remove the `prettier` rule in this repo in the next step.

**Comment by dotansimha** [comment]
Available in `v1.0.0`! 🎉
